app/util/rect.ts

```diff
@@ -0,0 +1,66 @@
+import { v2, Vec2 } from './vec2';
+import { ServiceHelper } from '../services/stateful-service';
+
+/**
+ * Class for a simple rectangle
+ */
+@ServiceHelper()
```

Comment:
I think we need to take a long hard look at `ServiceHelper` and figure out what it's future is.  We haven't really used it for much since its initial implementation.

It's strange to me that this random utility class would be using a decorator from `StatefulService` when this has nothing to do with mutations or the store.  Reading the comment on the `ServiceHelper` decorator doesn't help much, as it just talks about the fact that it can have mutations.

I'm assuming you're using this because you need to return an instance of this class from a service method and access the result in another window.  However, I think the `ServiceHelper` in its current form inherently confusing and difficult to use.  It's unclear what is actually required of classes that use this decorator.  For example, it looks like all `ServiceHelper` classes need to define a `getModel` method, but it's unclear what this should do and what it is used for.  It's also unclear to me what happens to any state on the instance that isn't recreated by constructor arguments.  I assume that just gets subtly lost when moving to a new window.  There's also a lot that happens with proxies with ServiceHelpers which seem to not be used or needed here.

Essentially what I'm saying is... until we've spent more time looking into the future of `ServiceHelper` and educating the team about it, I don't think we should be using it.  In this case you seem to only be using it to share some very simple behavior.  I don't think you gain much from having a `ServiceHelper` here.  You could simply have your service method return an `IRectangle` and instantiate this utility class in the child window.

Reply:
ServiceHelper allows to safely use this class in different windows
Also, it allows to construct this class via API calls and call all its methods.
You, right I need to add more clarification for this pattern for other developers in our team.

I just double checked the code, we can omit this decorator for the `Rect` class without bad consequences